### PR TITLE
Fuji api

### PR DESCRIPTION
**Pull request by arturrez**
ref: https://github.com/ava-labs/avalanche-cli/issues/1942
```
STATUS FOR CLUSTER: p0
======================

+---------------------+------------------------------------------+----------------+---------+-----------+---------------+------------------+-----------+
|      CLOUD ID       |                 NODE ID                  |       IP       | NETWORK |   ROLE    | AVAGO VERSION | PRIMARY NETWORK  |  HEALTHY  |
+---------------------+------------------------------------------+----------------+---------+-----------+---------------+------------------+-----------+
| i-0895050811460aac3 | NodeID-6fLEB9XMLa2CpBAjwHjL2CxKJEKdxiWi8 | -----------    | Fuji    | Validator | v1.11.8       | NOT_BOOTSTRAPPED | UNHEALTHY |
+---------------------+------------------------------------------+----------------+---------+-----------+---------------+------------------+-----------+
| i-09413f21fb5e58ab4 | NodeID-NH9KktuiJgfVYAqYRDNCJuNHxNU5u8gHY | -----------    | Fuji    | Validator | v1.11.8       | NOT_BOOTSTRAPPED | UNHEALTHY |
+---------------------+------------------------------------------+----------------+---------+-----------+---------------+------------------+-----------+
| i-0eea86f62df05f0ae | NodeID-56tASNv7or8yW4PWciwAqEME72mbBibpM | -----------    | Fuji    | API       | v1.11.8       | NOT_BOOTSTRAPPED | UNHEALTHY |
+---------------------+------------------------------------------+----------------+---------+-----------+---------------+------------------+-----------+
| i-0fc4d68effb04bf5f | NodeID-6qNtauyag8vGRDdQY7Fz6shVvB17CDUUP | -----------    | Fuji    | API       | v1.11.8       | NOT_BOOTSTRAPPED | UNHEALTHY |
+---------------------+------------------------------------------+----------------+---------+-----------+---------------+------------------+-----------+
| i-0c05fe6041f4bcf89 |                                          | -----------    | Fuji    | Monitor   |               |                  |           |
+---------------------+------------------------------------------+----------------+---------+-----------+---------------+------------------+-----------+
```

### CODE REVIEW

**Review by felipemadero — COMMENTED** [review]
nothing is changed in node conf for fuji API role. expected some avalanchego config
settings to be changed: http host key at least, and probably pchain only sync . most probably there were no check that the API node rpcs are publicly available

**Comment by arturrez** [comment]
added `--public-api` flag

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -68,6 +68,8 @@ const (
 	AWSGP3DefaultThroughput     = 125
 	SimulatePublicNetwork       = "SIMULATE_PUBLIC_NETWORK"
 
+	PublicAPI          = true
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/constants/constants.go:71]: added Visibility type

File: pkg/docker/config.go [hunk 2]
```diff
@@ -9,10 +9,14 @@ import (
...
 	"github.com/ava-labs/avalanche-cli/pkg/models"
 	"github.com/ava-labs/avalanche-cli/pkg/remoteconfig"
+	"github.com/ava-labs/avalanche-cli/pkg/utils"
 )
 
-func prepareAvalanchegoConfig(host *models.Host, networkID string) (string, string, error) {
-	avagoConf := remoteconfig.PrepareAvalancheConfig(host.IP, networkID, nil)
+func prepareAvalanchegoConfig(host *models.Host, network models.Network, isAPIHost bool) (string, string, error) {
+	avagoConf := remoteconfig.PrepareAvalancheConfig(host.IP, network.ShortID(), nil)
+	if network.PublicAPI || isAPIHost || utils.IsE2E() {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/docker/config.go:17]: replaced with publicAccess

File: pkg/models/network.go [hunk 3]
```diff
@@ -41,21 +41,23 @@ type Network struct {
 	ID          uint32
 	Endpoint    string
 	ClusterName string
+	PublicAPI   bool
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/models/network.go:44]: k

File: pkg/docker/config.go [hunk 4]
```diff
@@ -9,10 +9,14 @@ import (
 	"github.com/ava-labs/avalanche-cli/pkg/constants"
 	"github.com/ava-labs/avalanche-cli/pkg/models"
 	"github.com/ava-labs/avalanche-cli/pkg/remoteconfig"
+	"github.com/ava-labs/avalanche-cli/pkg/utils"
 )
 
-func prepareAvalanchegoConfig(host *models.Host, networkID string) (string, string, error) {
-	avagoConf := remoteconfig.PrepareAvalancheConfig(host.IP, networkID, nil)
+func prepareAvalanchegoConfig(host *models.Host, network models.Network, isAPIHost bool) (string, string, error) {
+	avagoConf := remoteconfig.PrepareAvalancheConfig(host.IP, network.ShortID(), nil)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/docker/config.go:16]: i think we should use NetworkIDFlagValue - as it was used before for configuration.
i removed ShortID - we don't need it

File: pkg/docker/ssh.go [hunk 5]
```diff
@@ -25,7 +25,7 @@ func ValidateComposeFile(host *models.Host, composeFile string, timeout time.Dur
 }
 
 // ComposeSSHSetupNode sets up an AvalancheGo node and dependencies on a remote host over SSH.
-func ComposeSSHSetupNode(host *models.Host, network models.Network, avalancheGoVersion string, withMonitoring bool) error {
+func ComposeSSHSetupNode(host *models.Host, network models.Network, avalancheGoVersion string, withMonitoring bool, isAPI bool) error {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/docker/ssh.go:28]: publicAccessToHTTPPort is a good name for this imho

File: pkg/networkoptions/networkoptions.go [hunk 6]
```diff
@@ -97,6 +97,9 @@ func AddNetworkFlagsToCmd(cmd *cobra.Command, networkFlags *NetworkFlags, always
 	if addEndpoint {
 		cmd.Flags().StringVar(&networkFlags.Endpoint, "endpoint", "", "use the given endpoint for network operations")
 	}
+	if addPublicAPI {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/networkoptions/networkoptions.go:100]: k

File: pkg/models/network.go [hunk 7]
```diff
@@ -140,6 +146,17 @@ func (n Network) NetworkIDFlagValue() string {
 	return "invalid-network"
 }
 
+func (n Network) ShortID() string {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/models/network.go:149]: agree. fixed

File: pkg/networkoptions/networkoptions.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/networkoptions/networkoptions.go:100]: moved to cloudconfig

File: pkg/ssh/ssh.go [hunk 8]
```diff
@@ -520,6 +520,10 @@ func RunSSHRenderAvalancheNodeConfig(app *application.Avalanche, host *models.Ho
 	if genesisFileExists(host) {
 		avagoConf.GenesisPath = filepath.Join(constants.DockerNodeConfigPath, constants.GenesisFileName)
 	}
+	// make sure that wiz works. TODO switch to RPC over SSH client to access 127.0.0.1 instead
```
Comment (felipemadero) [inline pkg/ssh/ssh.go:523]: let's just keep all devnet nodes open without ssh rpc. it is much better for ux experience and not needed to hide behind ssh for devnet

Comment (felipemadero) [reply pkg/ssh/ssh.go:523]: if possible remove the TODO

File: pkg/models/clustersConfig.go [hunk 9]
```diff
@@ -28,6 +28,7 @@ type ClusterConfig struct {
 	ExtraNetworkData   ExtraNetworkData
 	Subnets            []string
 	External           bool
+	HTTPAccess         constants.Visibility
```
Comment (felipemadero) [inline pkg/models/clustersConfig.go:31]: pls change Visibility type to something giving more idea of the context, RemoteHostAccess, or smb like that

File: pkg/constants/constants.go [hunk 10]
```diff
@@ -6,6 +6,8 @@ import (
 	"time"
 )
 
+type Visibility bool
```
Comment (felipemadero) [inline pkg/constants/constants.go:9]: pls change the type name to something like HostRemoteAccess or something like that

**Review by felipemadero — COMMENTED** [review]
after this, happily approving

File: pkg/models/clustersConfig.go (hunk 9 above)
Comment (arturrez) [reply pkg/models/clustersConfig.go:31]: changed to PublicHTTPAccess

File: pkg/constants/constants.go (hunk 10 above)
Comment (arturrez) [reply pkg/constants/constants.go:9]: changed to PublicHTTPAccess

File: pkg/ssh/ssh.go (hunk 8 above)
Comment (arturrez) [reply pkg/ssh/ssh.go:523]: agree

Comment (arturrez) [reply pkg/ssh/ssh.go:523]: in the sdk we will have a more liable way once RPC client is ready

**Review by felipemadero — APPROVED** [review]

**Pull request merged**